sql/sql_acl.cc

```diff
@@ -13183,14 +13183,19 @@ struct MPVIO_EXT :public MYSQL_PLUGIN_VIO
 /**
   a helper function to report an access denied error in most proper places
 */
-static void login_failed_error(THD *thd)
+static void login_failed_error(THD *thd, uint err_override = ER_ACCESS_DENIED_ERROR)
 {
-  my_error(access_denied_error_code(thd->password), MYF(0),
+  uint err = access_denied_error_code(thd->password);
+
+  if(err_override != ER_ACCESS_DENIED_ERROR)
+    err = err_override;
```

Comment:
Is this more readable?
`err= (err_override == ER_ACCESS_DENIED_ERROR) ? access_denied_error_code(thd->password) : err_override;`

alternately set `err= access_denied_error_code(` in `else` branch.

note style - no space on lhs of assignment.